### PR TITLE
[bitnami/kafka] Usage of Host IPs for external service without autodiscovery

### PR DESCRIPTION
**Pull request by sveba**
Description of the change

HOST_IP environment variable is dependent from externalAccess.autoDiscovery.enabled This is not right. This is fix for that issue.

Benefits

Fix the issue

Possible drawbacks

Applicable issues

Additional information

Checklist

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

### CODE REVIEW

**Comment by sveba** [comment]
this is a result of the discussion for CR https://github.com/bitnami/charts/pull/5963

**Comment by sveba** [comment]
We need this rather quick. Can someone pls review this PR.

**Comment by alemorcuq** [comment]
> this is a result of the discussion from CR #5963

Thanks for the PR, @sveba! This links seems to be invalid (it points to this PR). Can I get a link to the discussion to have more context about this change?

**Comment by sveba** [comment]
@alemorcuq sorry, I fixed the link

File: bitnami/kafka/templates/statefulset.yaml [hunk 1]
```diff
@@ -200,13 +200,11 @@ spec:
             {{- if .Values.externalAccess.autoDiscovery.enabled }}
             - name: SHARED_FILE
               value: "/shared/info.txt"
-            {{- if eq .Values.externalAccess.service.type "NodePort" }}
+            {{- end }}
             - name: HOST_IP
               valueFrom:
                 fieldRef:
                   fieldPath: status.hostIP
```
Comment (alemorcuq) [inline bitnami/kafka/templates/statefulset.yaml:207]: Let's keep this as it was before: only if the service type is NodePort.

**Review by alemorcuq — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks good, but I have one minor request.

**Comment by sveba** [comment]
@alemorcuq done

File: bitnami/kafka/templates/statefulset.yaml [hunk 2]
```diff
@@ -200,13 +200,13 @@ spec:
             {{- if .Values.externalAccess.autoDiscovery.enabled }}
             - name: SHARED_FILE
               value: "/shared/info.txt"
-            {{- if eq .Values.externalAccess.service.type "NodePort" }}
+            {{- end }}
+            {{- if eq .Values.externalAccess.service.useHostIPs "NodePort" }}
```
Comment (alemorcuq) [inline bitnami/kafka/templates/statefulset.yaml:204]:
```suggestion
            {{- if eq .Values.externalAccess.service.type "NodePort" }}
```

**Review by alemorcuq — CHANGES_REQUESTED** [review]

**Review by alemorcuq — APPROVED** [review]

**Pull request merged**